src/shared/components/home/donation_dialog.tsx

```diff
@@ -0,0 +1,87 @@
+import { Component } from "inferno";
+import { GetSiteResponse } from "lemmy-js-client";
+import * as sanitizeHtml from "sanitize-html";
+import { HttpService, I18NextService } from "../../services";
+import { donateLemmyUrl } from "../../config";
+
+interface Props {
+  site?: GetSiteResponse;
+}
+
+interface State {
+  show: boolean;
+}
+
+export class DonationDialog extends Component<Props, State> {
+  state: State = { show: this.initializeShow() };
+
+  constructor(props: any, context: any) {
+    super(props, context);
+    this.clickDonate = this.clickDonate.bind(this);
+    this.clickHide = this.clickHide.bind(this);
+    this.hideDialog = this.hideDialog.bind(this);
+  }
+
+  initializeShow(): boolean {
+    const lastNotifDate = new Date(
+      this.props.site?.my_user?.local_user_view.local_user
+        .last_donation_notification ?? Number.MAX_SAFE_INTEGER,
+    );
+
+    const oneYearAgo = new Date();
+    oneYearAgo.setFullYear(oneYearAgo.getFullYear() - 1);
+    return lastNotifDate < oneYearAgo;
+  }
+
+  render() {
+    if (this.state.show) {
+      return (
+        <div class="alert border-info bg-light-subtle fade show" role="alert">
+          <div class="d-flex">
+            <h4 class="alert-heading flex-grow-1">
+              {I18NextService.i18n.t("donation_dialog_title")}
+            </h4>
+            <button
+              type="button"
+              class="btn-close position-relative"
+              onClick={this.clickHide}
+              aria-label={I18NextService.i18n.t("donation_dialog_button_hide")}
+            ></button>
+          </div>
+          <div
+            class="card-text"
+            dangerouslySetInnerHTML={{
+              __html: sanitizeHtml(
+                I18NextService.i18n.t("donation_dialog_message"),
+              ),
+            }}
```

Comment:
Should get rid of raw html here similar to [joinlemmy pr](https://github.com/LemmyNet/joinlemmy-site/pull/467/files#diff-bc0ae13d62de3da6e730cdd077d2dcea98df71eaf724377c285b3955b1055698R43)

Reply:
I see that `donation_dialog_message` has html in it. It'd be better to remove that entirely and either: 

- Split it into 2 strings, IE `donation_dialog_message_1` and `donation_dialog_message_2`
- Replace the `<br>`s with `<1>` , as the i18next react uses.

The first is probably better and much easier. Really html shouldn't be used for line breaks, but only for within-sentence formatting where you need to use links, italics, etc.

Reply:
Alright got it.